src/Microsoft.TemplateEngine.Edge/BuiltInManagedProvider/GlobalSettingsTemplatePackageProvider.cs

```diff
@@ -207,6 +210,21 @@ public void Dispose()
             _globalSettings.Dispose();
         }
 
+        private async Task UpdateTemplatePackagesMetadataAsync(IEnumerable<IManagedTemplatePackage> templatePackages, CancellationToken cancellationToken)
+        {
+            _ = templatePackages ?? throw new ArgumentNullException(nameof(templatePackages));
+            var updatedPackages = new List<TemplatePackageData>(templatePackages.Select(tp => ((ISerializableInstaller)tp.Installer).Serialize(tp)));
+            var cachedPackages = await _globalSettings.GetInstalledTemplatePackagesAsync(cancellationToken).ConfigureAwait(false);
+
+            var updatedCachePackages = cachedPackages.Select(cp =>
+            {
+                var updatedPackage = updatedPackages.FirstOrDefault(up => up.MountPointUri == cp.MountPointUri);
+                return updatedPackage ?? cp;
+            }).ToArray();
```

Comment:
using dictionary based on `MountPointUri` might be more efficient (but not sure).